activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1664,6 +1664,8 @@ def table_name_matches?(from)
         /(?:\A|(?<!FROM)\s)(?:\b#{table_name}\b|#{quoted_table_name})(?!\.)/i.match?(from.to_s)
       end
 
+      FIELD_SQL_FUNC_REGEX = /(\b(?<field>field\((?>[^)(]+|\g<0>)*\))(\s+(?<direction>asc|desc)\b)?)/i
```

Comment:
I'm pretty sceptic at handling SQL with regexp. Anyway since FIELD is per my understanding MySQL function, it would be good idea to scope this for MySQL adapter only somehow if any contributor would welcome this change.

Reply:
Ahh I didn't realize it was MySQL-specific, good call. I'll look at existing code to see how other DBMS-specific handling is done. Thanks!

Reply:
I suggest to wait for some contributor reaction first. If I remember well there was some effort to try stop parsing SQL with regexp in the history of ActiveRecord at all. I'm not sure what's current state.

Reply:
> scope this for MySQL adapter only somehow

So I see other references to `FIELD` in query_methods.rb that don't seem to be gated behind a "if MySQL" check, such as #in_order_of. It's possible my change could be extended to suit any function by not looking for the name "field" specifically. 🤔 The logic would be the same regardless of the function, I believe: swap the ASC/DESC that follows the function call. Would that be preferable? I opted to handle just `FIELD` because it felt safest to target a specific use case rather than the general case.

Otherwise, I wonder if some override of #reverse_order in a gem like https://github.com/github/activerecord-trilogy-adapter might be preferable than a change here in rails/rails.